enumerables.rb

```diff
@@ -0,0 +1,227 @@
+      end
+      false
+    elsif block_given?
+      elements.my_each do |i|
+        return true if yield i
+      end
+      false
+    elsif !block_given? && !default
+      elements.my_each do |i|
+        return true if i
+      end
+      false
+    elsif default.class != Regexp && default.class != Class
+      elements.my_each do |i|
+        return true if i == default
+      end
+      false
+    end
+  end
+
+  def my_none?(default = nil)
+    elements = to_a
+    if default == Numeric
+      elements.my_each do |i|
+        unless i.class == Integer || i.class == Float || i.class == Complex
+          return true
+        end
+      end
+      false
+    elsif default.class == Class
+      elements.my_each do |i|
+        return false if i.class == default
+      end
+      true
+    elsif default.class == Regexp
+      elements.my_each do |i|
+        return false if i =~ default
+      end
+      true
+    elsif block_given?
+      elements.my_each do |i|
+        return false if yield i
+      end
+      true
+    elsif !block_given? && !default
+      elements.my_each do |i|
+        return false if i
+      end
+      true
+    elsif default.class != Regexp && default.class != Class
+      elements.my_each do |i|
+        return false if i == default
+      end
+      true
+    end
+  end
+
+  def my_count(default = nil)
+    elements = to_a
+    count = 0
+    if default
+      elements.my_each do |i|
+        count += 1 if i == default
+      end
+    elsif block_given?
+      elements.my_each do |i|
+        count += 1 if (yield i) == true
+      end
+    else
+      elements.my_each do
+        count += 1
+      end
+    end
+    count
+  end
+
+  def my_inject(first_argg = nil, sym = nil)
+    elements = to_a
+    if first_argg && sym
+      arg = sym
+      accumulate = first_argg
+      elements.my_each_with_index do |element, _i|
+        accumulate = accumulate.send(arg, element)
+      end
+      return accumulate
+    elsif first_argg && !sym
+      arg = first_argg
+    elsif !first_argg && !sym
+      accumulate = nil
+      elements.my_each do |element|
+        accumulate = if accumulate.nil?
+                       element
+                     else
+                       yield accumulate, element
+          end
+      end
+    end
+    if arg.class == Symbol
+      accumulate = nil
+      elements.my_each do |element|
+        accumulate = if accumulate.nil?
+                       element
+                     else
+                       accumulate.send(arg, element)
+          end
+      end
+    elsif block_given? && arg
+      accumulate = arg
+      elements.my_each do |element|
+        accumulate = yield accumulate, element
+      end
+    end
+    accumulate
+  end
+
+  def my_map(proc = nil)
+    return to_enum unless block_given?
+
+    elements = to_a
+
+    array = []
+    block_given? ?
+      elements.my_each { |i| array << proc.call(i) } :
+      elements.my_each { |i| array << yield(i) }
```

Comment:
Check these lines of code! You already know there is a block given, maybe you should check the state of the proc.... 🤓